### PR TITLE
Allow partial match of the container Id like shorter versio...

### PR DESCRIPTION
**Pull request by brahmaroutu**
...n from docker ps

Addresses #10645

Signed-off-by: Srini Brahmaroutu srbrahma@us.ibm.com

### CODE REVIEW

File: integration-cli/docker_cli_events_test.go [hunk 1]
```diff
@@ -293,3 +293,87 @@ func TestEventsFilters(t *testing.T) {
...
+	events = events[:len(events)-1]
+	if len(events) != 3 {
+		t.Fatalf("Expected 3 events, got %d: %v", len(events), events)
+	}
+	dieEvent = strings.Fields(events[len(events)-1])
+	if dieEvent[len(dieEvent)-1] != "die" {
+		t.Fatalf("event should be die, not %#v", dieEvent)
+	}
+
+	logDone("events - filters using container id")
```
Comment (tiborvass) [inline integration-cli/docker_cli_events_test.go]: please remove this logDone

File: events/events.go [hunk 2]
```diff
@@ -112,11 +113,15 @@ func writeEvent(job *engine.Job, event *utils.JSONMessage, eventFilters filters.
 			if v == field {
 				return false
 			}
+			if partialMatch && strings.Contains(field, v) {
```
Comment (tiborvass) [inline events/events.go]: short version is not partial match. Short version would be using `strings.HasPrefix` instead.

**Comment by tiborvass** [comment]
Also to fix that issue, I believe this PR should also handle container names.

**Comment by brahmaroutu** [comment]
@tiborvass I fixed the two issues mentioned above. I was thinking that adding the container name to the filter is a enhancement request and should be a separate PR? Let me know.

**Comment by tiborvass** [comment]
@brahmaroutu sorry for the delay. Your PR does handle short IDs, but if you would like to fix the issue you're referencing, it would have to also handle container names. Please note that container names have to fully match, not just partially.

I believe the easiest way to achieve this would be to use `daemon.Get(containerName)`, since that already handles correctly the ID short/long version + container name.

**Comment by brahmaroutu** [comment]
@tiborvass events' job are not registered from Daemon, I cannot get reference to daemon to call Get.

**Comment by tiborvass** [comment]
Closing since #10663 handles container names in addition to container IDs.

**Pull request closed without merging**